gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsMetricsMonitor.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.clustermonitor;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpUriBuilder;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.Response;
+import io.airlift.http.client.ResponseHandler;
+import io.airlift.http.client.UnexpectedResponseException;
+import io.airlift.log.Logger;
+import io.trino.gateway.ha.config.BackendStateConfiguration;
+import io.trino.gateway.ha.config.MonitorConfiguration;
+import io.trino.gateway.ha.config.ProxyBackendConfiguration;
+import io.trino.gateway.ha.security.util.BasicCredentials;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static io.airlift.http.client.ResponseHandlerUtils.propagate;
+import static io.trino.gateway.ha.clustermonitor.MonitorUtils.shouldRetry;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class ClusterStatsMetricsMonitor
+        implements ClusterStatsMonitor
+{
+    public static final String RUNNING_QUERIES_METRIC = "trino_execution_name_QueryManager_RunningQueries";
+    public static final String QUEUED_QUERIES_METRIC = "trino_execution_name_QueryManager_QueuedQueries";
+    private static final Logger log = Logger.get(ClusterStatsMetricsMonitor.class);
+    private final HttpClient client;
+    private final int retries;
+    private final MetricsResponseHandler metricsResponseHandler;
+    private final Header identityHeader;
+    private final String metricsEndpoint;
+    private final ImmutableSet<String> metricNames;
+    private final Map<String, Float> metricMinimumValues;
+    private final Map<String, Float> metricMaximumValues;
+
+    public ClusterStatsMetricsMonitor(HttpClient client, BackendStateConfiguration backendStateConfiguration, MonitorConfiguration monitorConfiguration)
+    {
+        this.client = requireNonNull(client, "client is null");
```

Comment:
NIT: more descriptive name like trinoMetricsClient would be better.

Reply:
I prefer shorter name here because there is no other clients in this class.

Reply:
this has been updated to `httpClient` to match the type and signature